src/app/leaves/actions/actions.ts

```diff
@@ -0,0 +1,27 @@
+"use server";
+import { z } from "zod";
+import { getSupabaseClient } from "../../../../utils/supabase/supabaseClient";
+import { leaveRequestSchema } from "../add/components/LeaveRequestFormBuilder";
+import { redirect } from "next/navigation";
+import { LeaveRequestStatus } from "@/constant/constant";
+
+export const onCreateLeaveRequest = async (
```

Comment:
Change this name too. We only add `on` prefix with functions which are event handlers or callbacks.